pkg/scheduler/admission/podnodeconstraints/admission_test.go

```diff
@@ -0,0 +1,133 @@
+package podnodeconstraints
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	_ "github.com/openshift/origin/pkg/api/install"
+	"github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints/api"
+
+	admission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+func emptyConfig() *api.PodNodeConstraintsConfig {
+	return &api.PodNodeConstraintsConfig{}
+}
+
+func testConfig(aa bool, bb bool) *api.PodNodeConstraintsConfig {
+	return &api.PodNodeConstraintsConfig{
+		ProhibitNodeName:     aa,
+		ProhibitNodeSelector: bb,
+	}
+}
+
+func defaultPod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	return pod
+}
+
+func nodeNamePod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Spec.NodeName = "frank"
+	return pod
+}
+
+func nodeSelectorPod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Spec.NodeSelector = map[string]string{"bogus": "frank"}
+	return pod
+}
+
+func emptyNodeSelectorPod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Spec.NodeSelector = map[string]string{}
+	return pod
+}
+
+func nilNodeSelectorPod() *kapi.Pod {
```

Comment:
There's no difference between this one and defaultPod above. The NodeSelector will  be nil by default.